File: src/Collega.Domain/IdeaFields/EffectiveField.cs
using Collega.Domain.Fields;

namespace Collega.Domain.IdeaFields;

/// <summary>
/// A field resolved for an idea's type: the <see cref="FieldDefinition"/> to render plus its effective
/// required-ness for that type (SPEC/20-feature-idea-type-fields.md "Effective-field resolution").
/// </summary>
public sealed record EffectiveField(FieldDefinition Field, bool Required);

File: packages/domain/src/idea-fields/effective-field.ts
import type { FieldDefinition } from '../fields/index.js'

/**
 * A field resolved for an idea's type: the `FieldDefinition` to render plus its effective
 * required-ness for that type (SPEC/20-feature-idea-type-fields.md "Effective-field resolution").
 */
export type EffectiveField = {
  readonly field: FieldDefinition
  readonly required: boolean
}
